backend/api/endpoints/hint.py

```diff
@@ -0,0 +1,23 @@
+from fastapi import APIRouter, Response, Cookie
+from typing import Optional
+
+from api.logic import user_management
+from routes import Hint
+from db_funcs import hintFuncs
+
+router = APIRouter()
+
+@router.post("/api/hint/{hint_id}")
+def get_hint(hint_id: int, response: Response, session_cookie: Optional[str] = Cookie(None)):
+    user = user_management.get_user_by_token(session_cookie)
+    if user is None:
+        response.status_code = 401
+        return {"result": "user is not authorized"}
+
```

Comment:
Tady chceme kontrolovat, jestli má uživatel právo na hint - tzn, jestli je v týmu, který je zaregistrovaný do hry, team vidí danou šifru, ke které se hint váže (tzn. vyřešil její prereq) a zároveň to je aktivní (není po deadlinu odevzdávání)